uint8 controller_addr
uint8 dev_addr
uint16 func_code
uint32 control_data
